src/sentry/replays/endpoints/project_replay_summarize_breadcrumbs.py

```diff
@@ -135,17 +135,34 @@ def fetch_error_details(project_id: int, error_ids: list[str]) -> list[GroupEven
                 category="error",
                 id=event_id,
                 title=data.get("title", ""),
-                timestamp=data.get("timestamp", 0.0) * 1000,  # error timestamp is in seconds
+                timestamp=data.get("timestamp", 0.0) * 1000,  # convert to milliseconds
                 message=data.get("message", ""),
             )
             for event_id, data in zip(error_ids, events.values())
-            if data is not None
+            if data is not None and data.get("timestamp") is not None
         ]
     except Exception as e:
         sentry_sdk.capture_exception(e)
         return []
 
 
+def parse_timestamp(timestamp_value: Any, unit: str) -> float:
+    """Parse a timestamp input to a float value.
+    The argument timestamp value can be string, float, or None.
+    The returned unit will be the same as the input unit.
+    """
+    if timestamp_value is not None:
+        if isinstance(timestamp_value, str):
+            try:
+                dt = datetime.fromisoformat(timestamp_value.replace("Z", "+00:00"))
+                return dt.timestamp() * 1000 if unit == "ms" else dt.timestamp()
```

Comment:
could just rename this func to be like `iso_to_ms(timestamp_value: Any) -> float` 
```suggestion
                return dt.timestamp() * 1000
```
then below it's like:
```
                timestamp = parse_timestamp(event.get("timestamp_ms")) or parse_timestamp(event.get("timestamp"))
```

Reply:
hrm, the `float(timestamp_value)` doesn't know about `unit`, so that could be a problem.